### ISSUE TITLE
(cpp) #define function

### ISSUE THREAD

**Issue by ngobach** [highlightjs/highlight.js#1362]
Hello. I have an issue when using highlight.js with this code:

```cpp
#include <bits/stdc++.h>
#define _for(i,a,b) for (int i=(a),_b_=(b);i<_b_;i++)
int main(){
    return 0;
}
```
highlight.js can't highlight the above code. But it work pefect when I remove the **#define** line.

Thanks you.

**Comment by yyyc514** [comment]
Please be specific both what you're expecting and an actual reproducible sample (jsfiddle, etc) always helps a lot.

![Screen Shot 2019-10-07 at 4 40 17 AM](https://user-images.githubusercontent.com/6473/66296932-9ef71780-e8bc-11e9-949a-29d2e71c0ca3.png)


It looks reasonable to me.  The only thing we really look for now in pre-processor directives is <> file paths... we don't try and process function like macros, etc.

But I'm not sure what you were seeing at the time.

**Comment by yyyc514** [comment]
Ping.

**Comment by ngobach** [comment]
Hi @yyyc514,

It happened to me with my project when I created this issue.
Here is an example of [bad snippet](https://paste.ngobach.com/view/-KXRMhP4Bn7fQy-v9lHX). And here is an example of [good snippet](https://paste.ngobach.com/view/-KXHJ5uOTE0TEK8NUudz).

However when I tried to reproduce this issue with latest version of highlight.js, both example are displayed correctly. So I think upgrading to latest version will help.

I think this issue can be closed now.
Thank you for your assist.
